postgres/datadog_checks/postgres/postgres.py

```diff
@@ -641,22 +641,23 @@ def _build_relations_config(self, yamlconfig):
         """
         config = {}
         for element in yamlconfig:
-            try:
-                if isinstance(element, str):
-                    config[element] = {'relation_name': element, 'schemas': []}
-                elif isinstance(element, dict):
-                    name = element['relation_name']
-                    config[name] = {}
-                    config[name]['schemas'] = element['schemas']
-                    config[name]['relation_name'] = name
-                else:
-                    self.log.warning('Unhandled relations config type: {}'.format(element))
-            except KeyError:
-                self.log.warning('Failed to parse config element={}, check syntax'.format(element))
+            if isinstance(element, str):
+                config[element] = {'relation_name': element, 'schemas': []}
+            elif isinstance(element, dict):
+                if 'relation_name' not in element or 'schemas' not in element:
+                    self.log.warning("Unknown element format for relation element %s", element)
+                    continue
+                if not isinstance(element['schemas'], list):
```

Comment:
I like adding this check but I feel the try/except block for KeyError is more future proof than checking if all keys are present